### ISSUE TITLE
Create basic components (Block, Link, Button, etc)

### ISSUE THREAD

**Issue by 1st** [1st/python-react#1]
To build extensible components - we need to have basic components (like Div or Block, Paragraph, Link, Button).

Implementing these basic components allows to go forward and make new components more faster.

**Comment by 1st** [comment]
I think that very base component should be Block, and all other components should extend it's functionality.

**Comment by alinarezrangel** [comment]
Is needed a specific GUI frontend, or the widgets should be abstracts for a possible graphical encapsulation later?

**Comment by 1st** [comment]
Hi @alinarezrangel.

Please find description on the README file on the main page of the project. I have described small example of possible usage.

Idea is similar to what ReactJS does - incapsulate representation and logic if tge component. Basic component used as a sceleton. Then we inherit it, for example, to render a Button. If it's a lemon link that stiled as a button - we render an A html tag. If it's not a link - we render it as an html Button.

Each component implements only own logic, allowing to build new UI components based on the parent component, extending his logic and representation.

**Comment by alinarezrangel** [comment]
Hi, I am now working on the classes (I will submit my pull request in a time).

**Comment by 1st** [comment]
Looks wery interesting. Any progress on it?

**Comment by alinarezrangel** [comment]
Yes, is almost ready, I used an abstract tree for represent the objects added and a rendering engine converts the XML-like tree to a HTML document.

**Comment by alinarezrangel** [comment]
It's done, I submitted my pull request. I try to encapsulate each component's logic into separated *"elements"* that can be rendered and extended by a *"rendering engine"*, I included also a **AbstractTree to HTML5** rendering engine for tests and uses in web servers. What do you think?
